docs/api.md

```diff
@@ -374,16 +374,6 @@ with_request_url("/users/42", host: "app.example.com") do
 end
 ```
 
-### `#with_variant(variant)`
```

Comment:
This file is autogenerated with YARD, can you make sure the method and comments are removed from the source file as well?

Reply:
Ah, thanks for clarifying! I think I’m realizing this is more than just a docs change. 

So this is the `with_variant` test helper and still publicly available. It feels like it should’ve been deprecated and removed [alongside `Base#with_variant`](https://github.com/ViewComponent/view_component/pull/1041/files)?

I’ve added https://github.com/ViewComponent/view_component/pull/1912/commits/3e1ab7c3e63b501bbd2d230fa64128e07e6a0aec which removes the method, too. It looks like we have a bunch of tests still using that helper - shall I continue down this path of removal?